src/Search.hs

```diff
@@ -0,0 +1,57 @@
+module Search
+    ( searchBoards
+    ) where
+
+import Control.Monad (mzero)
+import Data.Maybe (fromMaybe)
+import Data.List as List
+
+import Data.Aeson
+import Data.ByteString.Lazy as BS
```

Comment:
`import qualified`? Just `import ... as` will import it as both unqualified and as `BS`, so you can disambiguate with `BS` if you need to. I've found that's kind of confusing, so I use `import qualified` to _only_ import it as `BS`.